hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -462,6 +467,21 @@ Connection connect(Object target, Function<Object, Connection> getOrConnectFunct
         } catch (ClientNotAllowedInClusterException e) {
             logger.warning("Exception during initial connection to " + target + ": " + e);
             throw e;
+        } catch (TargetDisconnectedException e) {
+            logger.warning("Exception during initial connection to " + target + ": " + e);
+            connectionProcessListener.remoteClosedConnection(addressProvider.apply(target));
```

Comment:
Why do we need the `addressProvider.apply(target)`? You call the `connectionProcessListener.attemptingToConnectToAddress` for the non-translated address but call the `remoteClosedConnection` with the translated one.